docs/design/ttl_scans.md

```diff
@@ -0,0 +1,79 @@
+# TTL scans
+
+## Summary
+
+Add an option to automatically delete old starboard scans, in this first version focus on vulnerability scans but in the long run we could add similar functionality to other reports as well.
+
+## Motivation
+
+In [537](https://github.com/aquasecurity/starboard/issues/537) we talk about a need to run nightly vulnerability scans of CVE:s.
+This way we can make sure to get new CVE reports for long time running pods as well.
+
+## Proposal
+
+Add a environment variable to the operator, for example `OPERATOR_TTL_VULNERABILITY_SCANNER=24h`, this way we can add other ttl values for other reports as well.
+Adding this environment variable would add a annotation to the generated VulnerabilityReport that we can look for.
+
+Using similar functionality as defined in [ttlAfterFinished](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) for cronjob seems to be the most versatile option.
+
+Just like the [ttlAfterFinished controller](https://github.com/kubernetes/kubernetes/blob/2a821d787b7f21fbb399b0a0bf6d3fa0c79f4293/pkg/controller/ttlafterfinished/ttlafterfinished_controller.go) we should add it as a separate controller.
+This way it will be easy to extend to other reports as well.
+
+If the operator restarts the queue will disappear since we don't hold any state so we would have to scan all the available reports and calculate
```

Comment:
We are using controller-runtime and its Reconciler abstraction. Can you elaborate how we're going to init the controller on startup? If we block the init for too longit won't scale with a number of VulnerabilityReports. On the other hand, I'm not sure we have "access" to work queue outside the Reconilation handler function.

Reply:
Spoken to a friend about this and he pointed out that we don't need to use the ttlAfterFinished solution.
Instead we can use the controller result. https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Result

When we get a reconcile thanks to a vulnerability have been created we can go in and check the annotation `starboard.aquasecurity.github.io/report-ttl` and do the calculation on how long it should live.
and then just do a RequeueAfter and the controller will manage this magic for us :) 

```.go
type Result struct {
	// Requeue tells the Controller to requeue the reconcile key.  Defaults to false.
	Requeue bool

	// RequeueAfter if greater than 0, tells the Controller to requeue the reconcile key after the Duration.
	// Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.
	RequeueAfter time.Duration
}
```

My understanding is that this also will be checked during startup of the controller if there is already existing reports.
Does this sound reasonable @danielpacak? If so I will update the document sometime after Christmas and see if I can create a PoC around it.

Reply:
If we could use RequestAfter duration and confirm that existing VulnerabilityReports get listed by the controller manager that would be the best clean solution.